### PR TITLE
update default aoai version

### PR DESCRIPTION
**Pull request by sonichi**
## Why are these changes needed?

Update the default Azure OpenAI version to 2023-12-01-preview

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: test/agentchat/test_function_call.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 @pytest.mark.skipif(skip, reason="openai not installed OR requested to skip")
 def test_eval_math_responses():
     config_list = autogen.config_list_from_models(
-        KEY_LOC, exclude="aoai", model_list=["gpt-4-0613", "gpt-3.5-turbo-0613", "gpt-3.5-turbo-16k"]
+        KEY_LOC, model_list=["gpt-4-0613", "gpt-3.5-turbo-0613", "gpt-3.5-turbo-16k"]
```
Comment (ekzhu) [inline test/agentchat/test_function_call.py:24]: In the tests it's better to be more clear which model is being used. E.g., having separate tests for each model provider and each model. This way we can quickly realize which one has compatibility issues as those model providers may change their API specs anytime.

Comment (sonichi) [reply test/agentchat/test_function_call.py:24]: Ideally yes, right now we don't have enough resources to do that.

Comment (davorrunje) [reply test/agentchat/test_function_call.py:24]: Is this test using the Azure endpoint? If so, a deployment on Azure cannot have "." in its name. `gpt-3.5-turbo-0613` and `gpt-3.5-turbo-16k` is probably `gpt-35-turbo-0613` and `gpt-35-turbo-16k`.

Comment (sonichi) [reply test/agentchat/test_function_call.py:24]: They are automatically converted to "gpt-35-..." inside OpenAIWrapper's `_process_for_azure()`

**Review by joshkyh — COMMENTED** [review]
Noted on
    # TODO: recover this test after rewriting after the new group chat api

Thanks sonichi!

**Review by ekzhu — APPROVED** [review]

**Review by davorrunje — APPROVED** [review]

File: test/agentchat/test_function_call.py (hunk 1 above)
Comment (ekzhu) [reply test/agentchat/test_function_call.py:24]: I like how we are hiding the complexity from our users. Can this cause bug in some cases? Not a focus of this PR.

**Pull request merged**